airflow/providers/google/common/hooks/base_google.py

```diff
@@ -190,6 +190,7 @@ class GoogleBaseHook(BaseHook):
         If set as a sequence, the identities from the list must grant
         Service Account Token Creator IAM role to the directly preceding identity, with first
         account from the list granting this role to the originating account.
+    :param impersonation_scopes: Optional list of scopes for impersonated account. Will override scopes from connection.
```

Comment:
https://github.com/apache/airflow/issues/33400 reported the problem on BigQuery, then why are you adding it to base google class?

Reply:
`impersonation` is a feature of gcp, it is theoretically possible that another gcp service would also need this (searching for 'gcp missing scope' gives examples of other services needing scope but not sure how many relevant for airflow use case). 

What I was trying to solve here was a narrowly scoped service account would be able to get impersonate another account and have bigger scope. I would agree though that the most common use case for this is bigquery accessing google sheets. I can switch it over to bigquery library if you think that is better

Reply:
Lets please limit to BigQuery otherwise it will make things harder for https://github.com/apache/airflow/issues/15933

Reply:
i'm glad that splitting providers is still be considered. just pushed the changes

Reply:
Composer team is on it https://github.com/apache/airflow/pull/37223#issuecomment-1964943387